File: 01-数据结构/01-数组/c++/LC 删除排序数组中的重复项.cpp
#include<vector>

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() < 1 ) return 0;
        int l = 0;
        int r = 0;

        while(r < nums.size()) {
            if (nums[r] != nums[l]) {
                swap(nums[++l],nums[r]);
            }
            r++;
        }
        return l+1;
    }
};

File: 01-数据结构/01-数组/go/LC 删除排序数组中的重复项.go
package main

func removeDuplicates(nums []int) int {

	if len(nums) == 0 {
		return 0
	}
    r, l := 0, 0
    for r < len(nums) {
        if nums[r] != nums[l] {
            l++
            nums[l], nums[r] = nums[r], nums[l]
        }
        r++
    }
    return l+1;
}